gui/freeadmin/models/fields.py

```diff
@@ -57,6 +59,28 @@ def to_python(self, value):
         return value
 
 
+class EncryptedDictField(models.Field):
+    empty_strings_allowed = False
+
+    def get_internal_type(self):
+        return "TextField"
+
+    def get_db_prep_value(self, value, connection, prepared=False):
+        if not value:
+            value = {}
+        return notifier().pwenc_encrypt(json.dumps(value))
+
+    def from_db_value(self, value, expression, connection, context):
+        return self.to_python(value)
+
+    def to_python(self, value):
+        if not value:
+            return {}
+        if isinstance(value, str):
+            return json.loads(notifier().pwenc_decrypt(value))
```

Comment:
@themylogin Just to be sure, have you tested what happens if pwenc_decrypt fails? If that wont break backup.credential.query altogether?